Server/MirObjects/PlayerObject.cs

```diff
@@ -7464,8 +7497,70 @@ public void TakeQuestItem(ItemInfo uItem, ushort count)
                     item.Count -= count;
                 break;
             }
-        }       
-        
+        }
+
+        //Point-to-point
+        public void MemoryLocation(string name, int color)
+        {
+            if (Info.MyTeleportInfo.Count >= Globals.MaxPositionMove)
+            {
+                Enqueue(new S.Awakening { result = 11 });
+                return;
+            }
+
+            bool hasPositionScroll = false;
+            for (int i = 0; i < Info.Inventory.Length; i++)
+            {
+                UserItem item = Info.Inventory[i];
+                if (item == null || item.Info.Type != ItemType.Scroll) continue;
+                if (item.Info.Shape == 202)
+                {
+                    hasPositionScroll = true;
+                    Enqueue(new S.DeleteItem { UniqueID = item.UniqueID, Count = 1 });
+                    Info.Inventory[i] = null;
+                }
+                break;
+            }
+            if (hasPositionScroll)
+            {
+                PlayerTeleportInfo info = new PlayerTeleportInfo();
+                info.Name = name;
+                info.MapName = CurrentMap.Info.FileName;
+                info.Location = Info.CurrentLocation;
+                info.ColorIndex = color;
+                Info.MyTeleportInfo.Add(info);
+                Enqueue(new S.PlayerTeleportList { Infos = Info.MyTeleportInfo });
+            }
+            else
+            {
+                Enqueue(new S.Awakening { result = 12 });
+                return;
+            }
+        }
+        //Point-to-point
+        public void PositionMove(int Index)
+        {
+
+            if (Account.Gold <= 3000)
```

Comment:
would be better to have the "3000" gold also added as a global variable and applied throughout the code.